### PR TITLE
Fixes #22237 - notification on success run

### PR DESCRIPTION
**Pull request by ares**
see how it works
record

ready for later customization, job_invocation would return different notification builder class set per feature

### CODE REVIEW

File: app/lib/actions/remote_execution/run_hosts_job.rb [hunk 1]
```diff
@@ -66,6 +66,12 @@ def run(event = nil)
         super unless event == Dynflow::Action::Skip
       end
 
+      def finalize(*args)
```
Comment (adamruzicka) [inline app/lib/actions/remote_execution/run_hosts_job.rb]: Nitpick: I don't think `finalize` can take any arguments

**Comment by adamruzicka** [comment]
Looks mostly good to me, left one nitpick inline, however I'm hitting one issue which doesn't appear on your gif where the notification dropdown menu shows just 'Details' instead of 'Job Details'

menu detail

File: db/seeds.d/50-notification_blueprints.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+blueprints = [
+  {
+    group: N_('Jobs'),
+    name: 'rex_job_succeeded',
+    message: N_('%{subject} has finished successfully'),
```
Comment (iNecas) [inline db/seeds.d/50-notification_blueprints.rb]: Nitpick: IMO the %{subject} should be quoted to better distinguish between static and dynamic part of the notification.

**Comment by iNecas** [comment]
Nice work in general. Mind adding some tests for this. We don't have the test file for this currently, but https://github.com/theforeman/foreman_remote_execution/blob/master/test/unit/actions/run_hosts_job_test.rb could be used for inspiration: I'm at your hand if help needed on this part.

**Comment by ares** [comment]
@iNecas added test and fixed both comments

**Comment by iNecas** [comment]
Running the tests here http://ci.theforeman.org/view/Plugins/job/test_plugin_pull_request/4564/

**Comment by ares** [comment]
Fixed the rubocop, rerun at http://ci.theforeman.org/view/Plugins/job/test_plugin_pull_request/4570/, expected to fail on colorful output broken cops which is fixed by #310

File: test/unit/actions/run_hosts_job_test.rb [hunk 3]
```diff
@@ -119,5 +119,14 @@ class RunHostsJobTest < ActiveSupport::TestCase
         planned.input[:concurrency_control][:level].wont_be_empty
       end
     end
+
+    describe 'notifications' do
+      it 'creates notification on sucess run' do
+        FactoryBot.create(:notification_blueprint, :name => 'rex_job_succeeded')
+        assert_difference 'NotificationRecipient.where(:user_id => targeting.user.id).count' do
```
Comment (iNecas) [inline test/unit/actions/run_hosts_job_test.rb]: Nice: wasn't aware of this assertion.

**Comment by iNecas** [comment]
Thanks @ares

**Pull request merged**